src/Sylius/Bundle/CoreBundle/DependencyInjection/Compiler/LazyCacheWarmupPass.php

```diff
@@ -26,17 +27,20 @@ public function process(ContainerBuilder $container)
     {
         $this->markServiceAsLazy($container, 'cmf_core.templating.helper');
         $this->markServiceAsLazy($container, 'cmf_create.rdf_type_factory');
+        $this->markServiceAsLazy($container, 'fos_oauth_server.server');
     }
 
     /**
      * @param ContainerBuilder $container
-     * @param $id
+     * @param string $id
      */
     private function markServiceAsLazy(ContainerBuilder $container, $id)
     {
-        if ($container->hasDefinition($id)) {
+        try {
```

Comment:
I am just curious.. what is the advantage/difference of doing the try/catch over the hasDefinition check? Thank you.